pkg/cassandra/config.go

```diff
@@ -103,6 +103,40 @@ func newConfig(apiConfig api.CassandraConfig, cassandraVersion string) (config,
 	return cfg, nil
 }
 
+func addEncryptionOptions(apiConfig *api.CassandraConfig, encryptionStoresSecrets EncryptionStoresPasswords, cassandraVersion string) api.CassandraConfig {
+	updatedConfig := apiConfig.DeepCopy()
+	if updatedConfig.CassandraYaml.ClientEncryptionOptions == nil && updatedConfig.CassandraYaml.ServerEncryptionOptions == nil {
+		return *updatedConfig
+	}
+
+	if updatedConfig.CassandraYaml.ClientEncryptionOptions != nil {
+		if updatedConfig.CassandraYaml.ClientEncryptionOptions.Enabled {
+			keystorePath := fmt.Sprintf("%s/%s", StoreMountFullPath("client", "keystore"), "keystore")
+			truststorePath := fmt.Sprintf("%s/%s", StoreMountFullPath("client", "truststore"), "truststore")
+			updatedConfig.CassandraYaml.ClientEncryptionOptions.Keystore = pointer.String(keystorePath)
+			updatedConfig.CassandraYaml.ClientEncryptionOptions.Truststore = pointer.String(truststorePath)
+			updatedConfig.CassandraYaml.ClientEncryptionOptions.KeystorePassword = &encryptionStoresSecrets.ClientKeystorePassword
+			updatedConfig.CassandraYaml.ClientEncryptionOptions.TruststorePassword = &encryptionStoresSecrets.ClientTruststorePassword
+		}
+	}
+	if updatedConfig.CassandraYaml.ServerEncryptionOptions != nil {
+		if *updatedConfig.CassandraYaml.ServerEncryptionOptions.Enabled {
+			keystorePath := fmt.Sprintf("%s/%s", StoreMountFullPath("server", "keystore"), "keystore")
+			truststorePath := fmt.Sprintf("%s/%s", StoreMountFullPath("server", "truststore"), "truststore")
+			updatedConfig.CassandraYaml.ServerEncryptionOptions.Keystore = pointer.String(keystorePath)
+			updatedConfig.CassandraYaml.ServerEncryptionOptions.Truststore = pointer.String(truststorePath)
+			updatedConfig.CassandraYaml.ServerEncryptionOptions.KeystorePassword = &encryptionStoresSecrets.ServerKeystorePassword
+			updatedConfig.CassandraYaml.ServerEncryptionOptions.TruststorePassword = &encryptionStoresSecrets.ServerTruststorePassword
+		}
+		// The encryption stores shouldn't end up in the cassandra yaml, they are specific to k8ssandra
+		if IsCassandra3(cassandraVersion) {
```

Comment:
I'm not sure but it seems that `Optional` is not available for Cassandra 3 either.

Reply:
I don't fully understand what you're saying. Did you mean Cassandra 4?

Reply:
I was meaning that `optional` is defined in 4.0.1:

https://github.com/apache/cassandra/blob/cassandra-4.0.1/conf/cassandra.yaml#L1109

But not in 3.11.11:

https://github.com/apache/cassandra/blob/cassandra-3.11.11/conf/cassandra.yaml#L1045-L1057

But the code looks correct, I might have misread this.